### PR TITLE
Make node name unique

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java [hunk 1]
```diff
@@ -292,6 +305,39 @@ public UUID getStudyRootNodeUuid(UUID studyId) {
...
+        return nodesRepository.findAllByStudyId(studyUuid)
+                .stream()
+                .filter(nodeEntity -> nodeEntity.getType() == NodeType.NETWORK_MODIFICATION)
+                .map(nodeEntity -> networkModificationNodeInfoRepository.findById(nodeEntity.getIdNode()).get().getName())
+                .collect(Collectors.toList());
+    }
+
+    public String getUniqueNodeName(UUID studyUuid) {
+        String namePrefix = "New node ";
+        int counter = 0;
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java:322]: Maybe you should have a static counter here, to not restart at 0 each time you create a node...

File: src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java [hunk 2]
```diff
@@ -694,5 +701,36 @@ private void assertModificationNodeEquals(AbstractNode expected, AbstractNode cu
...
         assertEquals(expectedModificationNode.getBuildStatus(), currentModificationNode.getBuildStatus());
     }
+
+    @Test
+    public void testNodeName() throws IOException {
+        RootNode root = createRoot();
+        String nodeName = "node 1";
+        final NetworkModificationNode node = buildNetworkModification(nodeName, "potamus", null, VARIANT_ID, LoadFlowStatus.RUNNING, loadFlowResult2, UUID.randomUUID(), BuildStatus.BUILT);
+        createNode(root.getStudyId(), root, node);
+        node.setId(null);
```
Comment (sBouzols) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java:711]: Maybe create a `createNodeFail` method here which could be reused in other tests

**Review by sBouzols — CHANGES_REQUESTED** [review]
Mainly I see an unoptimized code here

File: src/main/java/org/gridsuite/study/server/networkmodificationtree/repositories/NetworkModificationNodeInfoRepository.java [hunk 3]
```diff
@@ -9,8 +9,12 @@
...
 import org.gridsuite.study.server.networkmodificationtree.entities.NetworkModificationNodeInfoEntity;
 
+import java.util.List;
+import java.util.UUID;
+
 /**
  * @author Jacques Borsenberger <jacques.borsenberger at rte-france.com
  */
 public interface NetworkModificationNodeInfoRepository extends NodeInfoRepository<NetworkModificationNodeInfoEntity> {
+    List<NetworkModificationNodeInfoEntity> findAllByNodeStudyId(UUID studyUuid);
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/study/server/networkmodificationtree/repositories/NetworkModificationNodeInfoRepository.java:19]: `findAllNodesByStudyUuid` ?

**Review by sBouzols — CHANGES_REQUESTED** [review]

**Review by sBouzols — APPROVED** [review]
Code review ok, will test with front

**Comment by sBouzols** [comment]
Look at the Code smell !

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java [hunk 4]
```diff
@@ -437,23 +423,25 @@ private void assertChildrenEquals(Set<AbstractNode> original, List<AbstractNode>
     @Test
     public void testNodeInsertion() throws Exception {
         RootNode root = createRoot();
-        final NetworkModificationNode networkModification = buildNetworkModification("hypo", "potamus", UUID.randomUUID(), VARIANT_ID, LoadFlowStatus.RUNNING, loadFlowResult2, UUID.randomUUID(), BuildStatus.BUILT_INVALID);
+        final NetworkModificationNode networkModification1 = buildNetworkModification("hypo 1", "potamus", UUID.randomUUID(), VARIANT_ID, LoadFlowStatus.RUNNING, loadFlowResult2, UUID.randomUUID(), BuildStatus.BUILT_INVALID);
```
[the remark this answers is not in the document]
Comment (YenguiSeddik) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java:426]: Each node must have a unique name to be created

File: src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java [hunk 5]
```diff
@@ -335,8 +320,9 @@ public void testNodeCreation() throws Exception {
     @Test
     public void testNodeManipulation() throws Exception {
         RootNode root = createRoot();
-        final NetworkModificationNode node1 = buildNetworkModification("hypo", "potamus", UUID.randomUUID(), VARIANT_ID, LoadFlowStatus.NOT_DONE, loadFlowResult, UUID.randomUUID(), BuildStatus.NOT_BUILT);
+        final NetworkModificationNode node1 = buildNetworkModification("hypo 1", "potamus", UUID.randomUUID(), VARIANT_ID, LoadFlowStatus.NOT_DONE, loadFlowResult, UUID.randomUUID(), BuildStatus.NOT_BUILT);
```
[the remark this answers is not in the document]
Comment (YenguiSeddik) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java:323]: Each node must have a unique name to be created

File: src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java [hunk 6]
```diff
@@ -322,6 +332,38 @@ public UUID getStudyRootNodeUuid(UUID studyId) {
...
+    }
+
+    private List<String> getStudyNodesNames(UUID studyUuid) {
+        return networkModificationNodeInfoRepository.findAllByNodeStudyId(studyUuid)
+                .stream()
+                .map(node -> node.getName())
+                .collect(Collectors.toList());
+    }
+
+    public String getUniqueNodeName(UUID studyUuid) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java:354]: add @Transactional(readOnly = true)
and move code getStudyNodesNames inside this method

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java (hunk 6 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java]: ok, done

**Review by SlimaneAmar — APPROVED** [review]

**Review by sBouzols — APPROVED** [review]

**Pull request merged**